### DISCUSSION TITLE
using oak and @graphql/yoga

### DISCUSSION THREAD

**Discussion by dagda1** [oakserver/oak#553] [Q&A]
I'd like to use oak and [@graphql-yoga](https://github.com/dotansimha/graphql-yoga).

At the moment, I am using it with `http/server.ts`:

```ts
import { serve } from 'http/server.ts'
import { createServer } from '@graphql-yoga/common'

const graphQLServer = createServer({
  schema: {
    typeDefs,
    resolvers
  },
  context: makeContext()
})

serve(graphQLServer.handleRequest, {
  port: PORT,
});
```

[Here](https://github.com/dotansimha/graphql-yoga/blob/e598de7e51620d649bf961d4a34313af306bd755/packages/common/src/server.ts#L550) `handleRequest` function.

Would it be possible to use oak with `@graphql-yoga`.

**Comment by kitsonk** [comment]
Currently you would habe to write your own wrapper to adapt the handler to oak middleware, but it feels like we should make it easier to use foreign handlers with oak.

**Reply by juslin03** [reply]
hey @kitsonk,
Any update of this ?

**Reply by kitsonk** [reply]
No.
